client/src/pages/board/components/dashboardForms/CardDialog.js

```diff
@@ -0,0 +1,209 @@
+import React, {useReducer, useState, useCallback, useEffect} from 'react';
+import {
+  Box,
+  Dialog,
+  DialogActions,
+  Button,
+  Divider,
+  MuiThemeProvider,
+  CircularProgress,
+  makeStyles,
+  DialogContent,
+} from '@material-ui/core';
+import {MuiPickersUtilsProvider} from '@material-ui/pickers';
+import DateFnsUtils from '@date-io/date-fns';
+import {getCardById} from 'api/Card';
+import CenteringBox from 'components/CenteringBox';
+import CardDialogTitle from './dialogSections/CardDialogTitle';
+import {dialogTheme} from '../../../../themes/theme';
+import CardDialogContentBody from './CardDialogContentBody';
+import SectionInfos from './dialogSections/enums';
+import CardDialogButtonMenu from './dialogSections/CardDialogButtonMenu';
+
+const useStyles = makeStyles({
+  content: {
+    display: 'flex',
+    alignItems: 'flex-start',
+    justifyContent: 'space-between',
+  },
+});
+
+const detailsReducer = (details, updatedSection) => {
+  return {...details, ...updatedSection};
+};
```

Comment:
I think sina mentioned it an earlier standup, but it seems Reducers aren't that necessary if you're just gonna basically replace your details with another version of details + the newSection. I think setState({...details, ...updatedSection}) works just as well.

Reply:
While this is true, [based on the docs' suggestion](https://reactjs.org/docs/hooks-reference.html#usereducer), I felt `useReducer` fit better for what I'm doing here. After all `useState` hook is an abstraction of `useReducer`.